inference-engine/samples/speech_sample/fileutils.cpp

```diff
@@ -0,0 +1,145 @@
+    if (in_file.good()) {
+        while (!in_file.eof()) {
+            std::string line;
+            uint32_t numRows = 0u, numCols = 0u, num_bytes = 0u;
+            std::getline(in_file, line, '\0');  // read variable length name followed by space and NUL
+            std::getline(in_file, line, '\4');  // read "BFM" followed by space and control-D
+            if (line.compare("BFM ") != 0) {
+                break;
+            }
+            in_file.read(reinterpret_cast<char*>(&numRows), sizeof(uint32_t));  // read number of rows
+            std::getline(in_file, line, '\4');                                  // read control-D
+            in_file.read(reinterpret_cast<char*>(&numCols), sizeof(uint32_t));  // read number of columns
+            num_bytes = numRows * numCols * sizeof(float);
+            in_file.seekg(num_bytes, in_file.cur);  // read data
+
+            if (numArrays == numArrayToFindSize) {
+                numMemoryBytes += num_bytes;
+            }
+            numArrays++;
+        }
+        in_file.close();
+    } else {
+        throw std::runtime_error(std::string("Failed to open %s for reading in GetFileInfo()!\n") + fileName);
+    }
+
+    if (ptrNumArrays != NULL)
+        *ptrNumArrays = numArrays;
+    if (ptrNumMemoryBytes != NULL)
+        *ptrNumMemoryBytes = numMemoryBytes;
+}
+
+void ArkFile::LoadFile(const char* fileName, uint32_t arrayIndex, std::string& ptrName, std::vector<uint8_t>& memory, uint32_t* ptrNumRows,
+                       uint32_t* ptrNumColumns, uint32_t* ptrNumBytesPerElement) {
+    std::ifstream in_file(fileName, std::ios::binary);
+    if (in_file.good()) {
+        uint32_t i = 0;
+        while (i < arrayIndex) {
+            std::string line;
+            uint32_t numRows = 0u, numCols = 0u;
+            std::getline(in_file, line, '\0');  // read variable length name followed by space and NUL
+            std::getline(in_file, line, '\4');  // read "BFM" followed by space and control-D
+            if (line.compare("BFM ") != 0) {
+                break;
+            }
+            in_file.read(reinterpret_cast<char*>(&numRows), sizeof(uint32_t));  // read number of rows
+            std::getline(in_file, line, '\4');                                  // read control-D
+            in_file.read(reinterpret_cast<char*>(&numCols), sizeof(uint32_t));  // read number of columns
+            in_file.seekg(numRows * numCols * sizeof(float), in_file.cur);      // read data
+            i++;
+        }
+        if (!in_file.eof()) {
+            std::string line;
+            std::getline(in_file, ptrName, '\0');  // read variable length name followed by space and NUL
+            std::getline(in_file, line, '\4');     // read "BFM" followed by space and control-D
+            if (line.compare("BFM ") != 0) {
+                fprintf(stderr, "Cannot find array specifier in file %s in LoadKaldiArkArray()!\n", fileName);
+                exit(-1);
```

Comment:
still observe exit codes

Reply:
Thanks, I missed it. Now it is exception.